examples/oneapi/cpp/source/pca/pca_dense_online.cpp

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/pca.hpp"
+#include "oneapi/dal/io/csv.hpp"
+
+#include "example_util/utils.hpp"
+
+namespace dal = oneapi::dal;
+
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/pca.hpp"
+#include "oneapi/dal/io/csv.hpp"
+
+#include "example_util/utils.hpp"
+
+namespace dal = oneapi::dal;
+
+template <typename Method>
+void run(const dal::table& x_train, const std::string& method_name) {
+    const auto pca_desc =
+        dal::pca::descriptor<float, Method>().set_component_count(5).set_deterministic(true);
+    const std::int64_t nBlocks = 10;
+
+    dal::pca::partial_train_result<> partial_result;
+    std::cout << method_name << "\n" << std::endl;
+    auto input_table = split_table_by_rows<double>(x_train, nBlocks);
+
```

Comment:
Good point, will be added in separate pr

Reply:
I also think that it would be better not to split the data table which was read from a single file into 10 blocks. More naturally would be to have several  (maybe not 10, but rather 3) input files that contain the input blocks of data. And compute the result from those files.

Reply:
Yes, this comment is related to examples only.
In Bazel tests it is more convenient to have the data generated on the fly.